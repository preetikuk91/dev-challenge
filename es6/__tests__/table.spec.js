import Table from "../table";
import Data  from "../testData.json";
describe("test calculateData", () => {
  test("it should render table", () => {
    const testData = { body: JSON.stringify(Data)};

    document.body.innerHTML = "<div id=\"resp-table-body\">\n" +
      "            <div id=\"resp-table-row\"></div>\n" +
      "        </div>";
    window.Sparkline = {draw: jest.fn()};
    const t = new Table();
    jest.spyOn(t, 'addData');
    jest.spyOn(t, 'appendChild');
    jest.spyOn(t, 'sortSparkLineData');
    jest.spyOn(t, 'renderData');
    jest.spyOn(t, 'getData');
    t.addData(testData);
    expect(t.data).toMatchObject({});
    expect(t.data).toHaveProperty(Data.name);
    expect(t.addData).toHaveBeenCalledTimes(1);
    expect(t.addData).toHaveBeenCalledWith(testData);
    expect(t.sortSparkLineData).toHaveBeenCalled();
    expect(t.appendChild).toHaveBeenCalled();
    expect(t.renderData).toHaveBeenCalled();
    expect(t.getData).toHaveBeenCalled();
    expect(document.getElementsByClassName('table-body-cell')).toBeTruthy();
    expect(document.getElementsByClassName('table-body-cell')).toHaveLength(6);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

});