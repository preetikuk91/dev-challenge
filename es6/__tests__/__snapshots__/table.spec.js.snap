// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calculateData it should render table 1`] = `"<div id=\\"resp-table-body\\"><div class=\\"resp-table-row\\"><div class=\\"table-body-cell\\">usdjpy</div><div class=\\"table-body-cell\\">106.7297012204255</div><div class=\\"table-body-cell\\">107.25199883791178</div><div class=\\"table-body-cell\\">-2.8769211401569663</div><div class=\\"table-body-cell\\">-4.862314256927661</div><div class=\\"table-body-cell\\"><span></span></div></div></div>"`;
